internal/resources/cluster/resource_cluster.go

```diff
@@ -295,68 +298,117 @@ func resourceClusterDelete(_ context.Context, d *schema.ResourceData, m interfac
+			cluster.Spec.TkgVsphere.Distribution.Version = newVersion.(string)
 
-	updateRequired := false
+			log.Printf("[INFO] updating TKGm vSphere workload cluster version to %s", newVersion.(string))
 
-	switch {
-	case common.HasMetaChanged(d):
-		fallthrough
-	case d.HasChange(helper.GetFirstElementOf(SpecKey, clusterGroupKey)):
-		updateRequired = true
+			return true
+		}
 	}
 
-	if !updateRequired {
-		return diags
-	}
+	return false
+}
 
-	getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
-	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
+func withTKGsVsphereVersionUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) bool {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey)) {
+		newVersion := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey))
+		if newVersion.(string) != "" {
+			cluster.Spec.TkgServiceVsphere.Distribution.Version = newVersion.(string)
+
+			log.Printf("[INFO] updating TKGs workload cluster version to %s", newVersion.(string))
+
+			return true
+		}
 	}
 
-	if common.HasMetaChanged(d) {
-		meta := common.ConstructMeta(d)
+	return false
+}
+
+func withClusterGroupUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) bool {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, clusterGroupKey)) {
+		newClusterGroupName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+		if newClusterGroupName.(string) != "" {
+			cluster.Spec.ClusterGroupName = newClusterGroupName.(string)
+
+			log.Printf("[INFO] updating cluster group to %s", newClusterGroupName.(string))
 
-		if value, ok := getResp.Cluster.Meta.Labels[common.CreatorLabelKey]; ok {
-			meta.Labels[common.CreatorLabelKey] = value
+			return true
 		}
+	}
+
+	return false
+}
 
-		getResp.Cluster.Meta.Labels = meta.Labels
-		getResp.Cluster.Meta.Description = meta.Description
+func withMetaUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) bool {
+	if !common.HasMetaChanged(d) {
+		return false
 	}
 
-	incomingCGName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+	objectMeta := common.ConstructMeta(d)
 
-	if incomingCGName.(string) != "" {
-		getResp.Cluster.Spec.ClusterGroupName = incomingCGName.(string)
+	if value, ok := cluster.Meta.Labels[common.CreatorLabelKey]; ok {
+		objectMeta.Labels[common.CreatorLabelKey] = value
 	}
 
-	_, err = config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceUpdate(
-		&clustermodel.VmwareTanzuManageV1alpha1ClusterRequest{
-			Cluster: getResp.Cluster,
-		},
-	)
+	cluster.Meta.Labels = objectMeta.Labels
+	cluster.Meta.Description = objectMeta.Description
+
+	log.Printf("[INFO] updating meta data")
```

Comment:
updating cluster metadata. 

You have used cluster in all the logs statement so please keep it consistent

Reply:
Will take up in the next PR